app/cisc/cisc.go

```diff
@@ -0,0 +1,312 @@
+	case 2:
+		// We'll get all arguments after
+	case 3:
+		name = c.Args().Get(2)
+	default:
+		log.Fatal("Please give the following arguments: group.toml id [hostname]", c.NArg())
+	}
+	group := getGroup(c)
+	idBytes, err := hex.DecodeString(c.Args().Get(1))
+	log.ErrFatal(err)
+	id := identity.ID(idBytes)
+	clientApp = identity.NewIdentity(group.Roster, 2, name)
+	clientApp.AttachToIdentity(id)
+	log.ErrFatal(saveConfig())
+	return nil
+}
+func idFollow(c *cli.Context) error {
+	log.Fatal("Not yet implemented")
+	return nil
+}
+func idRemove(c *cli.Context) error {
+	log.Fatal("Not yet implemented")
+	return nil
+}
+func idCheck(c *cli.Context) error {
+	log.Fatal("Not yet implemented")
+	return nil
+}
+
+/*
+ * Commands related to the config in general
+ */
+func configUpdate(c *cli.Context) error {
+	assertCA()
+	log.ErrFatal(clientApp.ConfigUpdate())
+	log.ErrFatal(clientApp.ProposeFetch())
+	log.Info("Successfully updated")
+	return configList(c)
+}
+func configList(c *cli.Context) error {
+	assertCA()
+	log.Info("Account name:", clientApp.ManagerStr)
+	log.Infof("Identity-ID: %x", clientApp.ID)
+	log.Infof("Current config: %s", clientApp.Config)
+	if c.Bool("p") {
+		if clientApp.Proposed != nil {
+			log.Infof("Proposed config: %s", clientApp.Proposed)
+		} else {
+			log.Info("No proposed config")
+		}
+	}
+	return nil
+}
+func configPropose(c *cli.Context) error {
+	assertCA()
+	log.Fatal("Not yet implemented")
+	return nil
+}
+func configVote(c *cli.Context) error {
+	assertCA()
+	log.ErrFatal(clientApp.ProposeVote(!c.Bool("r")))
+	return nil
+}
+
+/*
+ * Commands related to the key/value storage and retrieval
+ */
+func kvList(c *cli.Context) error {
+	assertCA()
+	log.Infof("config for id %x", clientApp.ID)
+	for k, v := range clientApp.Config.Data {
+		log.Infof("%s: %s", k, v)
+	}
+	return nil
+}
+func kvValue(c *cli.Context) error {
+	log.Fatal("Not yet implemented")
+	return nil
+}
+func kvAdd(c *cli.Context) error {
+	assertCA()
+	if c.NArg() < 2 {
+		log.Fatal("Please give a key value pair")
+	}
+	key := c.Args().Get(0)
+	value := c.Args().Get(1)
+	prop := clientApp.GetProposed()
+	prop.Data[key] = value
+	log.ErrFatal(clientApp.ProposeSend(prop))
+	return nil
+}
+func kvRm(c *cli.Context) error {
+	assertCA()
+	if c.NArg() != 1 {
+		log.Fatal("Please give a key to delete")
+	}
+	key := c.Args().First()
+	prop := clientApp.GetProposed()
+	if _, ok := prop.Data[key]; !ok {
+		log.Fatal("Didn't find key", key, "in the config")
+	}
+	delete(prop.Data, key)
+	log.ErrFatal(clientApp.ProposeSend(prop))
+	return nil
+}
+
+/*
+ * Commands related to the ssh-handling
+ */
+func sshAdd(c *cli.Context) error {
+	assertCA()
+	if c.NArg() != 1 {
+		log.Fatal("Please give the hostname as argument")
+	}
+
+	// Get the current configuration
+	sc, err := NewSSHConfigFromFile(sshConfig)
+	log.ErrFatal(err)
+
+	// Add a new host-entry
+	hostname := c.Args().First()
+	alias := c.String("a")
+	if alias == "" {
+		alias = hostname
```

Comment:
First argument has to be there, as it gives the hostname for which the ssh-key is valid. `if c.Narg() != 1` above.